modules/localization/msf/local_integ/measure_republish_process.cc

```diff
@@ -0,0 +1,480 @@
+    }
+    pthread_mutex_unlock(&height_mutex_);
+    measure_data.is_have_variance = true;
+  } else {
+    measure_data.measure_type = MeasureType::GNSS_POS_VEL;
+    measure_data.frame_type = FrameType::ENU;
+    measure_data.is_have_variance = false;
+
+    static bool is_sent_init_pos = false;
+    if (!is_sent_init_pos) {
+      is_sent_init_pos = true;
+      measure_data.gnss_vel.ve = 0.0;
+      measure_data.gnss_vel.vn = 0.0;
+      measure_data.gnss_vel.vu = 0.0;
+      LOG(INFO) << "Novatel bestgnsspose publish: "
+                << "send sins init position using novatel bestgnsspos!";
+      *measure = measure_data;
+      return true;
+    }
+
+    static int position_good_counter = 0;
+    if ((bestgnsspos_msg.longitude_std_dev() > 5.0) ||
+        (bestgnsspos_msg.latitude_std_dev() > 5.0)) {
+      LOG(WARNING) << "the position std is large: "
+                << bestgnsspos_msg.longitude_std_dev() << " "
+                << bestgnsspos_msg.latitude_std_dev();
+      return false;
+    }
+    if (position_good_counter < 10) {
+      ++position_good_counter;
+      pre_bestgnsspose_ = measure_data;
+      return false;
+    }
+
+    if (pre_bestgnsspose_.time) {
+      double re = 6378137.0;
+      double e = 0.0033528;
+      double temp_sin_lati_2 = sin(measure_data.gnss_pos.latitude) *
+                               sin(measure_data.gnss_pos.latitude);
+      double rn = re / (1 - e * temp_sin_lati_2);
+      double rm = re / (1 + 2.0 * e - 3.0 * e * temp_sin_lati_2);
+
+      double inv_time = 1.0 / (measure_data.time - pre_bestgnsspose_.time);
+      if (measure_data.time - pre_bestgnsspose_.time > 1.05) {
+        pre_bestgnsspose_ = measure_data;
+        position_good_counter = 0;
+        return false;
+      }
+      measure_data.gnss_vel.ve = (measure_data.gnss_pos.longitude -
+                                  pre_bestgnsspose_.gnss_pos.longitude) *
+                                 rn * inv_time *
+                                 cos(measure_data.gnss_pos.latitude);
+      measure_data.gnss_vel.vn = (measure_data.gnss_pos.latitude -
+                                  pre_bestgnsspose_.gnss_pos.latitude) *
+                                 rm * inv_time;
+      measure_data.gnss_vel.vu =
+          (measure_data.gnss_pos.height - pre_bestgnsspose_.gnss_pos.height) *
+          inv_time;
+
+      pre_bestgnsspose_ = measure_data;
+      LOG(INFO) << "novatel bestgnsspos velocity: "
+                << measure_data.gnss_vel.ve << " "
+                << measure_data.gnss_vel.vn << " "
+                << measure_data.gnss_vel.vu;
+
+      static double pre_yaw_from_vel = 0.0;
+      double yaw_from_vel =
+          atan2(measure_data.gnss_vel.ve, measure_data.gnss_vel.vn);
+      if (!pre_yaw_from_vel) {
+        pre_yaw_from_vel = yaw_from_vel;
+        return false;
+      } else {
+        double delta_yaw = yaw_from_vel - pre_yaw_from_vel;
+        if (delta_yaw > 6.1087) {
```

Comment:
use config for 6.1087

Reply:
Accrord to the contex, 6.1087 is changed to M_PI, and 6.2832 below is changed  to 2 *  M_PI.